kernel/src/common/deferred_call.rs

```diff
@@ -0,0 +1,49 @@
+//! Deferred call mechanism.
+//!
+//! This is a tool to allow chip peripherals to schedule "interrupts"
+//! in the chip scheduler if the hardware doesn't support interrupts where
+//! they are needed.
+
+use core::convert::Into;
+use core::convert::TryFrom;
+use core::convert::TryInto;
+use core::marker::Copy;
+use core::sync::atomic::AtomicUsize;
+use core::sync::atomic::Ordering;
+
+static DEFERRED_CALL: AtomicUsize = AtomicUsize::new(0);
+
+/// Are there any pending `DeferredCall`s
+pub fn has_tasks() -> bool {
+    DEFERRED_CALL.load(Ordering::Relaxed) != 0
+}
+
+/// Represents a way to generate an asynchronous call without a hardware interrupt.
+pub struct DeferredCall<T>(T);
+
+impl<T: Into<usize> + TryFrom<usize> + Copy> DeferredCall<T> {
```

Comment:
There's a requirement that the range of T is from 0 to 31, but that isn't enforced in this interface or documented anywhere.